.github/workflows/unit_tests.yml

```
@@ -311,7 +311,7 @@ jobs:
 
     - uses: actions/cache@v4
       with:
-        key: testing_data-1-${{ env.TESTING_VERSION }}
+        key: testing_data-1-${{ env.TESTING_VERSION }}-${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.mne-version }}
```

The testing data shouldn't depend on the OS or the python version
```suggestion
        key: testing_data-1-${{ env.TESTING_VERSION }}-${{ matrix.mne-version }}
```